### ISSUE TITLE
Changelog generation breaks with the [skip ci] tag

### ISSUE THREAD

**Issue by magicus** [Wynntils/Wynntils#2771]
As highlighted in https://github.com/Wynntils/Wynntils/issues/2769, the generated changelog looks weird when the PR title has a `[skip ci]` tag. It looks like this:
`Some real change skip ci] ([#1234)`
instead of this:
`Some real change [skip ci] (#1234)`

This just looks weird, and should be fixed. One way of fixing it (that I would recommend) is to actually remove all references to `[skip ci]` from PR titles when generating the changelog.

**Comment by magicus** [comment]
I see this has been targeted. I would actually propose two changes to address this:

1) Stop writing ´[skip ci]` on almost every PR. Instead, add `[release]` on the (much fewer) issues that we want to trigger a release. That is, invert the logic. This will both avoid the risk of inadvertently releasing on a simple fix due to a missed annotation, and it will cut down the amount of boilerplate, since we typically have several non-release PRs for each one that ends up triggering a release. 

This would entail a rather trivial change in the CI scripts.

2) Filter out this `[release]` tag from all subsequent processing. I think this means that it should be filtered out already in the CI scripts, since I believe that Athena is picking up the changelog from the files generated there. Otherwise we need to add additional filtering in Athena.

**Comment by ShadowCat117** [comment]
I plan to fix this with https://github.com/Wynntils/Wynntils/issues/3233, I'm not commited to any idea yet but the one I had was we just make prs as normal, then when we want to release we make a release pr that will update a changelog file that will be bundled with the mod and that will be published to the mod hosting sites and the releases tab here as well as being visible in game. Athena would have to be updated to get the new changelog, I've not looked into how that currently works, so that previous changelogs would still be visible

**Comment by magicus** [comment]
I appreciate that you want to write manual release documents. I'm just worried that this can be hard to keep up with, and that it might become a blocker. I'd very much prefer a situation where there at least exists an automatic changelog like we have now, if nothing else is done. But with a way to append/override with a custom changelog if that is something that we manages to do. Does that sound acceptable to you?

**Comment by ShadowCat117** [comment]
I've been writing manual release notes for the past 6 months on discord, there's a thread in the artemis-dev channel where I post them per pr once I've had the chance to read over them.

I think if after a new release was made we automatically generated a pr replacing the existing changelog file with a new one we can just edit that each time we merge a pr and then it just needs to be merged to do the release

**Comment by magicus** [comment]
Or we can have a special GHA script that we can run, like `Prepare new release`. This would create a new special release-PR, and incorporate the current changelog in the description. We can then edit the description, if we like, and when we then integrate it, it will cause the CI scripts to generate a release. I think that would be pretty nifty. 

It might even be possible to add Discord integration so that we have a discord command to trigger the  `Prepare new release` script to run.

**Comment by Rafii2198** [comment]
The PR idea additionally gives us and other people a chance to voice an opinion if something in suggested changelog could be improved. I also think github releases should follow the current changelog system, this one is much more insightful for developers as users should not come here anyway, they should download the mod from Curseforge or Modrinth, this gives them a more accurate changelog for their mods, as sometimes some change might be small while other time it's a full refactor, and it's not that reflected in manually written changelogs.

Basically a mix of both solutions, PRs would no longer trigger update by themselves, instead a special PR will be created for that where we can input and discuss a manually written one, when that gets merged, this manual changelog would be used on mod distribution sites and potentially posted on discord, while Github release would get the "old school" changelog that would be useful for developers. Also, that way there is still this organization of PRs with `feat:` `fix:` and so on which is useful to filter through PRs, tho that could be substituted with tags.

**Comment by magicus** [comment]
I think it sounds like we've targeted in on a good solution. Let's just check that @kristofbolyai is on-board with the idea as well, then I think we have a good consensus about this.

**Comment by kristofbolyai** [comment]
Yes, skip ci is bad. Adding release PRs is alright, but it might as well be just a manually triggerable github action (perhaps with the DC hook). However, do we have to have separate PRs for the changelog? Do we have to bundle them? This feels like something that should stay online, so if we want, we can change details about it. I see no problem with the mod fetching the changelog like it does now.

**Comment by magicus** [comment]
What is a DC hook?

**Comment by magicus** [comment]
I think it makes sense to bundle the changelog with the release, but if you want to keep the current online solution I'm okay with that too. I don't think there is much point in allowing us to change the changelog after the release. Even if we happened to write something bad, I just say "whatever" and leave it. It's just a changelog, after all.

**Comment by ShadowCat117** [comment]
I think we do need to have the changelog for a new release stored somewhere on the repo so that the publish to modrinth/curseforge can use it. Whether that is just a file for the repo to have or something that is actually included in the mod is up for debate, I guess it would be nice for the mod to have the latest changelog locally without needing to download it from athena.

I do agree that there's not much point in us being able to edit a changelog, most people will only ever read it once if at all

**Comment by magicus** [comment]
Oh, I see, "DC" is "Discord".
